slidedeck/src/slides/06-react-redux/00-react-redux.pug

```diff
@@ -0,0 +1,159 @@
+section
+  h1 React Redux
+
+section
+  h2 React Redux
+  ul
+    li Used to making binding React components to Redux a little easier
+
+section
+  h2 Tenets
+  ul
+    li Presentational (dumb) components vs. container (smart) components
+    li Avoid using state or React lifecycle methods in components if possible
+
+section
+  h3 Presentational components
+  ul
+    li Don't know about Redux
+    li Get data from props
+    li.
+      Can be implemented as <span class="component red">functional stateless components</span>
+
+section
+  h4 Example functional stateless component
+  script(type="text/plain", class="language-jsx", style="width: 100%; max-height: 700px;").
+    import React from 'react'
+    import PropTypes from 'prop-types'
+
+    const AgentDetails  = ( { agent, onClick } ) => (
+      <li onClick={onClick}>
+        {agent.name} - {agent.title}
+      </li>
+    )
+
+    AgentDetails.propTypes = {
+      onClick: PropTypes.func.isRequired,
+      agent: PropTypes.shape({
+        name: PropTypes.string.isRequired,
+        title: PropTypes.string.isRequired,
+      })
+    }
+
+    export default AgentDetails
+
+section
+  h3 Container components
+  ul
+    li Used as a bridge between React and Redux to dispatch actions and update the store
+    li.
+      Uses React Redux <span class="component red">connect()</span> function
+
+section
+  h3 Connect function
+  ul
+    li.
+      <span class="component red">connect([mapStateToProps], [mapDispatchToProps], [mergeProps], [options])</span>
+    li.
+      <span class="component red">mapStateToProps</span> and <span class="component red">mapDispatchToProps</span> are the most commonly used parameters
+
+section
+  h3 mapStateToProps(state, [ownProps])
+  ul
+    li Must be an object
```

Comment:
Or a function that returns an object...

Reply:
true. think it's worth covering? the react-redux docs make it sound like a pretty advanced use case.